infra/aws/terraform/prow-build-cluster/README.md

```diff
@@ -1,109 +1,250 @@
+
+* cluster name
+* cluster admin IAM role name
+* secrets manager IAM policy name
+* canary is missing k8s-prow OIDC provider and the corresponding role
+* subnet setup is different
+* instance type and autoscaling parameters (mainly for saving)
+
+## Interacting with clusters
+
+You'll mainly interact with clusters using kubectl and Terraform. You need
+kubeconfig for the former which can be obtained using the `aws` CLI.
+
+Production:
 
 ```bash
-# Fetch & update kubeconfig.
-# For Prod:
 aws eks update-kubeconfig --region us-east-2 --name prow-build-cluster
-# For Canary:
+```
+
+Canary:
+
+```bash
 aws eks update-kubeconfig --region us-east-2 --name prow-build-canary-cluster
+```
+
+This is going to update your `~/.kube/config` (unless specified otherwise).
+Once you fetched kubeconfig, you need to update it to add assume role arguments,
+otherwise you'll have no access to the cluster (e.g. you'll get Unauthorized
+error).
+
+Open kubeconfig in a text editor of your choice and update `args` for the
+appropriate cluster:
+
+* Production
+    ```yaml
+    args:
+      - --region
+      - us-east-2
+      - eks
+      - get-token
+      - --cluster-name
+      - prow-build-cluster
+      - --role-arn
+      - arn:aws:iam::468814281478:role/Prow-Cluster-Admin
+    ```
+* Canary:
+    ```yaml
+    args:
+      - --region
+      - us-east-2
+      - eks
+      - get-token
+      - --cluster-name
+      - prow-build-canary-cluster
+      - --role-arn
+      - arn:aws:iam::468814281478:role/canary-Prow-Cluster-Admin
+    ```
+
+## Running Terraform
+
+**WARNING: We strongly recommend using the provided Makefile to avoid
+mistakes due to selecting wrong environment!!!**
+
+We have a Makefile that can be used to execute Terraform targeting the
+appropriate/correct environment. This Makefile uses the following environment
+variables to control Terraform:
+
+* `WORKSPACE_NAME` (default: `canary`, can be `prod`)
+* `ASSUME_ROLE` (default: `true`) - whether to authenticate to AWS using
+  provided credentials or by assuming the ProwClusterAdmin role
+* `DEPLOY_K8S_RESOURCES` (default: `true`) - whether to deploy Kubernetes
+  resources defined via Terraform
+* `TF_ARGS` (default: none) - additional command-line flags and arguments
+  to provide to Terraform
+
+### Commands
+
+**WARNING: Make sure to read the whole document before creating a cluster
+for the first time as the additional steps are needed!**
+
+Init (**make sure to run this command before getting started with configs**):
+
+```bash
+make init
+```
+
+Plan:
 
-# create cluster role bindings
-kubectl apply -f ./resources/rbac
+```bash
+make plan
 ```
 
-Lastly, run Terraform script again without additinal variable. This time, it will implicitly assume
-previously created role and provision resources on top of EKS cluster.
+Apply:
 
 ```bash
 make apply
 ```
 
-From here, all consecutive runs should be possible with command from above.
+Destroy:
+
+```bash
+make destroy
```

Comment:
Is this equivalent to `TF_ARGS=-destroy make apply`?

Reply:
Yes, it should be. For the reference:

```
  -destroy               Destroy Terraform-managed infrastructure.
                         The command "terraform destroy" is a convenience alias
                         for this option.
```